general/mcp/cli.go

```
@@ -0,0 +1,443 @@
+}
+
+// Cmd handles the CLI command execution and argument parsing
+func Cmd(c *cli.Context) error {
+	// Show help if needed
+	if show, err := cliutils.ShowCmdHelpIfNeeded(c, c.Args()); show || err != nil {
+		return err
+	}
+	// Validate arguments
+	cmdArg := c.Args().Get(0)
+	switch cmdArg {
+	case "update":
+		return updateMcpServerExecutable()
+	case "start":
+		cmd := createAndConfigureCommand(c)
+		return commands.Exec(cmd)
+	default:
+		return cliutils.PrintHelpAndReturnError(fmt.Sprintf("Unknown subcommand: %s", cmdArg), c)
+	}
+}
+
+// updateMcpServerExecutable forces an update of the MCP server binary
+func updateMcpServerExecutable() error {
+	log.Info("Updating MCP server binary...")
+	osName, arch, binaryName := getOsArchBinaryInfo()
+	fullPath, exists, err := getLocalBinaryPath(binaryName)
+	if err != nil {
+		return err
+	}
+
+	var currentVersion string
+	// Check current version if binary exists
+	if exists {
+		currentVersion, err = getMcpServerVersion(fullPath)
+		if err != nil {
+			log.Warn("Could not determine current MCP server version:", err)
+		} else {
+			log.Info("Current MCP server version:", currentVersion)
+		}
+	}
+
+	// Check if we already have the latest version
+	if exists && currentVersion != "" {
+		latestVersion, err := getLatestMcpServerVersion(osName, arch)
+		switch {
+		case err != nil:
+			log.Warn("Could not determine latest MCP server version:", err)
+		case currentVersion == latestVersion:
+			log.Info("MCP server is already at the latest version:", currentVersion)
+			return nil
+		default:
+			log.Info("A newer version is available:", latestVersion)
+		}
+	}
+
+	// Download the latest version
+	_, err = downloadBinary(fullPath, defaultServerVersion, osName, arch)
+	if err != nil {
+		return err
+	}
+
+	// Check new version after update
+	newVersion, err := getMcpServerVersion(fullPath)
+	if err != nil {
+		log.Warn("Could not determine new MCP server version:", err)
```

Wouldn't we want to return an error in this case?